### PR TITLE
Save Correspondence Tasks & Upload to eFolder

### PR DESCRIPTION
**Pull request by tradesmanhelix**
Resolves [Save Correspondence Tasks & Upload to eFolder](https://jira.devops.va.gov/browse/APPEALS-36279)

# Description
Persists mail task data and "tasks not related to an appeal" data on correspondence intake form submit.

## Acceptance Criteria
- [ ] Code compiles correctly
- [ ] All specs passing

## Testing Plan
1. Fill out intake correspondence form, being sure to add data for "mail tasks" and "tasks not related to an appeal".
2. Verify the submit is successful and the above-mentioned data is persisted correctly.

# Frontend
## User Facing Changes
n/a

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [x] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [x] No new code climate issues added

### CODE REVIEW

File: app/services/correspondence_intake_processor.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      appeal_type: Correspondence.name,
+      type: CorrespondenceTask.name
+    )
+
+    parent = CorrespondenceRootTask.find_or_create_by!(
+      appeal_id: correspondence.id,
+      assigned_to: MailTeam.singleton,
+      appeal_type: Correspondence.name,
+      parent_id: correspondence_task.id,
+      type: CorrespondenceRootTask.name
```
Comment (zurbergram) [inline app/services/correspondence_intake_processor.rb:127]: This is incorrect.

CorrespondenceTask is an "abstract" parent class

CorrespondenceRootTask is created for a Correspondence in https://github.com/department-of-veterans-affairs/caseflow/blob/feature/APPEALS-27311/app/models/correspondence.rb after creation

File: client/app/queue/correspondence/correspondenceReducer/correspondenceReducer.js [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const initialState = {
   correspondences: [],
   radioValue: '0',
   relatedCorrespondences: [],
-  mailTasks: {},
+  mailTasks: [],
```
Comment (tradesmanhelix) [inline client/app/queue/correspondence/correspondenceReducer/correspondenceReducer.js:11]: Simplifying this to just be an array of the tasks names. When a box is checked, the corresponding mail task name gets added to the array. And it's removed when the box is unchecked.

This simplifies the frontend/backend code and is overall more in line with the way the rest of this data works.

File: spec/requests/correspondence_controller_spec.rb [hunk 3]
```diff
@@ -5,34 +5,34 @@
...
+          veteran_id: 3909
+        },
+        veteranInformation: {
+          id: 3909
+        },
+        waivedEvidenceTasks: []
+      }.to_json
+    end
+
     it "saves the user's current step in the intake form" do
```
Comment (jimfoley25) [inline spec/requests/correspondence_controller_spec.rb:36]: this test would not pass unless changed to 
`intake_correspondence = CorrespondenceIntake.find_by(user: current_user, correspondence: correspondence)
      expect(response).to have_http_status(:success)
      expect(intake_correspondence.current_step).to eq(current_step)
      expect(intake_correspondence.redux_store).to eq(redux_store)`
maybe caused by having a let value reused as a variable

Comment (tradesmanhelix) [reply spec/requests/correspondence_controller_spec.rb:36]: 👍 great catch, thank you!

**Review by jimfoley25 — APPROVED** [review]

**Pull request merged**